### ISSUE TITLE
`unfulfilled_lint_expectations` is incorrectly being triggered by `expect(clippy::let_and_return)`

### ISSUE THREAD

**Issue by AngheloAlf** [rust-lang/rust-clippy#16451]
### Summary

Running clippy from Rust 1.93 triggers an `unfulfilled_lint_expectations` on a `#[expect(clippy::let_and_return)]` that wasn't being triggered before.

This can be reproduced in the Rust Playground with this link: https://play.rust-lang.org/?version=stable&mode=debug&edition=2024&gist=a3d0ccb9433819399902e77c7ffe1fae

As a workaround changing the `#[expect(clippy::let_and_return)]` to `#[allow(clippy::let_and_return)]` makes the warning go away, but it is undesirable.

### Lint Name

unfulfilled_lint_expectations

### Reproducer

I tried this code:

```rust
use std::collections::BTreeMap;

pub fn test_func() -> BTreeMap<(), ()> {
    let stuff = BTreeMap::new();
    
    // TODO: Fill `stuff`

    #[expect(clippy::let_and_return)]
    stuff
}
```

I saw this happen:

```
    Checking playground v0.0.1 (/playground)
warning: this lint expectation is unfulfilled
 --> src/lib.rs:6:14
  |
6 |     #[expect(clippy::let_and_return)]
  |              ^^^^^^^^^^^^^^^^^^^^^^
  |
  = note: `#[warn(unfulfilled_lint_expectations)]` on by default

warning: `playground` (lib) generated 1 warning
    Finished `dev` profile [unoptimized + debuginfo] target(s) in 0.57s
```

I expected to see this happen:

No warning at all.

### Version

```text
rustc 1.93.0 (254b59607 2026-01-19)
binary: rustc
commit-hash: 254b59607d4417e9dffbc307138ae5c86280fe4c
commit-date: 2026-01-19
host: x86_64-unknown-linux-gnu
release: 1.93.0
LLVM version: 21.1.8
```
